components/renku_data_services/session/db.py

```diff
@@ -208,6 +209,17 @@ async def insert_launcher(
                         message=f"Session environment with id '{environment_id}' does not exist or you do not have access to it."  # noqa: E501
                     )
 
+            resource_class_id = new_launcher.resource_class_id
+            if resource_class_id is not None:
+                res = await session.scalars(
+                    select(schemas.ResourceClassORM).where(schemas.ResourceClassORM.id == resource_class_id)
+                )
+                resource_class = res.one_or_none()
+                if resource_class is None:
+                    raise errors.MissingResourceError(
+                        message=f"Resource class with id '{resource_class_id}' does not exist or you do not have access to it."  # noqa: E501
+                    )
+
```

Comment:
issue: we should use `get_classes` with the `user` here. doing DB queries in the session db on things owned by the crc DB doesn't separate concerns properly and despite the error saying `or you do not have access to it`, this does not actually check permissions. 
So here we should call ResourcePoolRepository.get_classes. For this to work, we need to inject the ResourcePoolRepository into the SessionRepository. similar to line 47 for project_authz.
To be clear, this is not something you did wrong, but something this whole file is guilty of and that we need to address in the future. But at least for new code we should do it properly.

Reply:
Sure, included in the latest commit.